core/Pos/Arbitrary/Crypto.hs

```diff
@@ -62,61 +49,33 @@ instance Arbitrary SignTag where
 -- Arbitrary signing keys
 ----------------------------------------------------------------------------
 
--- If you want an arbitrary keypair, just generate a secret key with
--- 'arbitrary' and then use 'Pos.Crypto.toPublic' to get the corresponding
--- public key.
-
-keys :: [(PublicKey, SecretKey)]
-keys = unsafeMakePool "[generating keys for tests...]" 50 keyGen
-{-# NOINLINE keys #-}
-
 instance Arbitrary PublicKey where
-    arbitrary = fst <$> elements keys
+    arbitrary = fst . deterministicKeyGen <$> arbitrarySizedS 32
 instance Arbitrary SecretKey where
-    arbitrary = snd <$> elements keys
-
-instance Nonrepeating PublicKey where
-    nonrepeating n = map fst <$> sublistN n keys
-instance Nonrepeating SecretKey where
-    nonrepeating n = map snd <$> sublistN n keys
-
--- Repeat the same for ADA redemption keys
-redemptionKeys :: [(RedeemPublicKey, RedeemSecretKey)]
-redemptionKeys = unsafeMakePool "[generating redemption keys for tests..]" 50 redeemKeyGen
+    arbitrary = snd . deterministicKeyGen <$> arbitrarySizedS 32
 
 instance Arbitrary RedeemPublicKey where
-    arbitrary = fst <$> elements redemptionKeys
+    arbitrary = fst . fromMaybe err . redeemDeterministicKeyGen <$>
+                arbitrarySizedS 32
+      where err = error "arbitrary@RedeemPublicKey: keygen failed"
```

Comment:
Why? Afaik there's no difference between `error` and `throw` in the `Gen` monad.